move es pillows to kafka only

emord opened the pull request:
Opening this up for tests. It'll move all pillows to kafka checkpoints

emord committed:
Add checkpoint handler to only use kafka checkpoints

Revert change to all elastic pillows

_get_checkpoints helper

emord committed:
Move all ES pillows to use kafka checkpoints

emord committed:
fix es pillow test case

emord committed:
use correct formats for checkpoint tests

emord committed:
fix test + comments

dimagimon commented on corehq/ex-submodules/pillowtop/checkpoints/manager.py:
```diff
@@ -150,6 +150,119 @@ def fire_change_processed(self, change, context):
+        return False
+
+
+class WrappedCheckpoint(object):
```
F811 redefinition of unused 'WrappedCheckpoint' from line 153

dimagimon commented on corehq/ex-submodules/pillowtop/checkpoints/manager.py:
```diff
@@ -150,6 +150,119 @@ def fire_change_processed(self, change, context):
+        return self.kafka_seq
+
+
+class KafkaPillowCheckpoint(PillowCheckpoint):
```
F811 redefinition of unused 'KafkaPillowCheckpoint' from line 164

emord committed:
bad merge

emord committed:
add partition only relies on kafka checkpoint

emord commented:
@snopoke I changed the management command to only use kafka checkpoint here https://github.com/dimagi/commcare-hq/pull/16473#discussion_r120386793

We still need to use django pillow checkpoint until this is merged, which i planned on doing monday once I get off interrupt.

snopoke commented:
not merging due to label

emord commented:
found a bug today that affects pillows with multiple topics. Will fix and merge on monday

emord committed:
do not add missing topics to a change feed

Because we are switching to partitioned pillows, the topics and
partitions will be determined before since. If we try to add more
topics, it will could add an intentionally ignored topic to the feed.

For example, case pillows should listen to case and case-sql. Assume
there is one partition for each and we create two pillows. The first
pillow should listen to case, second should listen to case-sql. Before
this change they would both listen to case and case-sql

emord committed:
partition case pillow

emord commented:
Added partitioning to the case pillow as I think that's all we need for right now.

snopoke commented on corehq/apps/change_feed/consumer/feed.py:
```diff
@@ -80,9 +80,6 @@ def iter_changes(self, since, forever):
             ]
-            topics_missing = set(self._topics) - checkpoint_topics
-            for topic in topics_missing:
-                offsets.append(topic)  # consume all available partitions
```
what's the reason for removing this?

emord replied:
documented in the commit message https://github.com/dimagi/commcare-hq/pull/16020/commits/2107e8ee5a7319712d77edc2dd6cd45ded751d0d

emord committed:
Remove change feed test for partial offsets

We shouldn't have partial offsets anymore

emord committed:
change case to es change feed type to partitioned

emord commented:
:white_check_mark:  @snopoke I think this is ready to go today provided I answered your question well
